subsys/caf/modules/ble_adv.c

```diff
@@ -498,23 +495,51 @@ static void ble_initialized(void)
 	started = true;
 }
 
+static inline void get_req_modules(struct module_flags *mf)
+{
+	BUILD_ASSERT(!IS_ENABLED(CONFIG_BT_SETTINGS) ||
+		     IS_ENABLED(CONFIG_CAF_SETTINGS_LOADER),
+		     "Bluetooth settings enabled, but settings loader module is disabled");
```

Comment:
For me `BUILD_ASSERT(!(IS_ENABLED(CONFIG_BT_SETTINGS) && !IS_ENABLED(CONFIG_CAF_SETTINGS_LOADER)), ...);` would be much more clear.

Reply:
Or simply:

```BUILD_ASSERT(IS_ENABLED(CONFIG_BT_SETTINGS) == IS_ENABLED(CONFIG_CAF_SETTINGS_LOADER), ...);```